Dockerfile

```diff
@@ -1,14 +1,23 @@
 FROM golang:1.4
 MAINTAINER Brian Tiger Chow <btc@perfmode.com>
 
+ENV IPFS_PATH /data/ipfs
+
+RUN useradd -m -d /data -u 1000 ipfs && \
+    mkdir -p /data/ipfs && chown ipfs:ipfs /data/ipfs
+
+EXPOSE 4001 5001 8080
+# 4001 = Swarm, 5001 = API, 8080 = HTTP transport
+
+VOLUME /data/ipfs
+
 ADD . /go/src/github.com/ipfs/go-ipfs
 RUN cd /go/src/github.com/ipfs/go-ipfs/cmd/ipfs && go install
 
 RUN cp /go/src/github.com/ipfs/go-ipfs/bin/container_daemon /usr/local/bin/start_ipfs && \
     chmod 755 /usr/local/bin/start_ipfs
 
-EXPOSE 4001 5001 8080
-# 4001 = Swarm, 5001 = API, 8080 = HTTP transport
+USER ipfs
```

Comment:
On Mon, Apr 27, 2015 at 08:09:33AM -0700, Knut Ahlers wrote:
> >  ADD . /go/src/github.com/ipfs/go-ipfs
> >  RUN cd /go/src/github.com/ipfs/go-ipfs/cmd/ipfs && go install
> >  
> >  RUN cp /go/src/github.com/ipfs/go-ipfs/bin/container_daemon /usr/local/bin/start_ipfs && \
> >      chmod 755 /usr/local/bin/start_ipfs
> >  
> > -EXPOSE 4001 5001 8080
> > -# 4001 = Swarm, 5001 = API, 8080 = HTTP transport
> > +USER ipfs
> 
> As the user `ipfs` does not own the `/go/src` and `/go/bin` dirs its
> not useful to change the user earlier than this.

Good point (the ‘ipfs’ user can't add to /go/src/…).  This PR looks
good to me as it stands.  The Travis failure is one of the sporadic
“No output has been received in the last ... minutes”, so I wouldn't
worry about that.  +1 to merge.